duckdb-vortex/src/vortex_scan.cpp

```diff
@@ -263,7 +263,10 @@ static void CreateScanPartitions(ClientContext &context, const BindData &bind, S
 
 	const auto partition_count = std::max(static_cast<uint64_t>(1), row_count / partition_size);
 	global_state.partitons_total += partition_count;
-	local_state.thread_local_file_idx = file_idx;
+
+	if (PinFileToThread(global_state)) {
```

Comment:
Why do you reeval this in each iteration below?

Reply:
Extracted this into a var if thats what you meant.